test/test.jl

```diff
@@ -227,6 +227,18 @@ end
 @test counter_17462_pre == 3
 @test counter_17462_post == 1
 
+# Issue #21008
+ts = try
+    @testset "@test_broken and @test_skip should not give an exception" begin
+        @test_broken false
+        @test_skip true
+        @test_skip false
+    end
+catch
+    nothing # Shouldn't get here
```

Comment:
Yes.  I can't remember exactly what I was thinking here, but it probably amounted to something like "`@test typeof(ts) == blah` is clearer than an `@test false` inside the catch".